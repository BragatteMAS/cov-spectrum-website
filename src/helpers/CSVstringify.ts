const CSVstringify = (data: any[]) => {
  let result = '';
  if (data.length > 0) {
    result += Object.keys(data[0]).join(',') + '\n';
    data.slice(1).forEach(item => {
      result += Object.values(item).join(',') + '\n';
    });
  }
  return result;
};

export default CSVstringify;
