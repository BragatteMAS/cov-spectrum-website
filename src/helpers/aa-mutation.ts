import { decodeNucMutation } from './nuc-mutation';
//import { ReferenceGenomeService } from '../services/ReferenceGenomeService';

import jsonRefData from '../data/refData.json';

const loadRefData = () => {
  // getting a 'clean' copy: https://stackoverflow.com/a/51681510
  const geneData = JSON.parse(JSON.stringify(jsonRefData));
  console.log(geneData);
  const startPositionsByGene = new Map<string, number>();
  for (const item of geneData.genes) {
    startPositionsByGene.set(item.name, item.startPosition);
  }
  console.log(startPositionsByGene);
  return startPositionsByGene;
};

export type DecodedAAMutation = {
  gene: string;
  originalBase: string | undefined;
  position: number;
  mutatedBase: string | undefined;
};

export function decodeAAMutation(mutation: string): DecodedAAMutation {
  const [gene, tail] = mutation.split(':');
  const decodedTail = decodeNucMutation(tail);
  return {
    gene,
    ...decodedTail,
  };
}

export function decodedAAMutationToString(mutation: DecodedAAMutation) {
  return (
    mutation.gene + ':' + (mutation.originalBase || '') + mutation.position + (mutation.mutatedBase || '')
  );
}

/**
 * A mutation list should be first sorted by the protein, then by the position within the protein.
 */
export function sortAAMutationList(mutations: string[]): string[] {
  return sortListByAAMutation(mutations, x => x);
}

export function sortListByAAMutation<T>(list: T[], mutationExtractorFunc: (x: T) => string) {
  const refData = loadRefData();

  return list
    .map(x => {
      const mutation = mutationExtractorFunc(x);
      const mutationDecoded = decodeAAMutation(mutation);
      return { x, mutationDecoded };
    })
    .sort((a, b) => {
      const startPositionA = refData.has(a.mutationDecoded.gene) ? refData.get(a.mutationDecoded.gene) : 0;
      const startPositionB = refData.has(b.mutationDecoded.gene) ? refData.get(a.mutationDecoded.gene) : 0;
      if (startPositionA && startPositionB) {
        if (startPositionA > startPositionB) {
          return 1;
        } else if (startPositionA < startPositionB) {
          return -1;
        } else {
          return 0;
        }
      } else {
        return 0;
      }
    })
    .map(m => m.x);
}

export function isValidAAMutation(mutation: string): boolean {
  // TODO Only allow existing genes
  return /^[A-Z]{1,3}[0-9]{0,2}[AB]?:[A-Z]?[0-9]+[A-Z-*\\.]?$/.test(mutation.toUpperCase());
}

export function isValidNspNotation(mutation: string): boolean {
  // TODO Only allow existing genes
  return /^NSP[0-9]{0,2}:[A-Z]?[0-9]+[A-Z]?$/.test(mutation.toUpperCase());
}

export function isValidABNotation(mutation: string): boolean {
  // TODO Only allow existing genes

  return /^ORF1AB:[A-Z]?[0-9]+[A-Z]?$/.test(mutation.toUpperCase());
}
