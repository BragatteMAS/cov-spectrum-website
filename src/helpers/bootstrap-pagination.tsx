import { Pagination } from 'react-bootstrap';
import React from 'react';

export type BootstrapPaginationControl = {
  elements: JSX.Element[];
};

type OnPageClickFunction = {
  (page: number): void;
};

function createItem(number: number, active: boolean, onPageClick: OnPageClickFunction) {
  return (
    <Pagination.Item key={number} active={active} onClick={() => onPageClick(number)}>
      {number}
    </Pagination.Item>
  );
}

/**
 * The page numbers start with 1.
 */
export function createBootstrapPaginationControl(
  totalElements: number,
  elementsPerPage: number,
  activePage: number,
  onPageClick: OnPageClickFunction
): BootstrapPaginationControl {
  const numberPages = Math.ceil(totalElements / elementsPerPage);
  let elements = [
    <Pagination.First key='first' onClick={() => onPageClick(1)} />,
    <Pagination.Prev key='previous' onClick={() => onPageClick(activePage - 1)} />,
  ];
  if (numberPages < 7) {
    for (let i = 1; i <= numberPages; i++) {
      elements.push(createItem(i, i === activePage, onPageClick));
    }
  } else {
    if (activePage !== 1) {
      elements.push(createItem(1, false, onPageClick));
    }
    if (activePage > 3) {
      elements.push(<Pagination.Ellipsis key='ellipsis-1' />);
    }
    if (activePage > 2) {
      elements.push(createItem(activePage - 1, false, onPageClick));
    }
    elements.push(createItem(activePage, true, onPageClick));
    if (activePage < numberPages - 1) {
      elements.push(createItem(activePage + 1, false, onPageClick));
    }
    if (activePage < numberPages - 2) {
      elements.push(<Pagination.Ellipsis key='ellipsis-2' />);
    }
    if (activePage !== numberPages) {
      elements.push(createItem(numberPages, false, onPageClick));
    }
  }

  elements.push(<Pagination.Next key='next' onClick={() => onPageClick(activePage + 1)} />);
  elements.push(<Pagination.Last key='last' onClick={() => onPageClick(numberPages)} />);
  return {
    elements,
  };
}
