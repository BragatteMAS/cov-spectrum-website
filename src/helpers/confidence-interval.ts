import { any } from 'zod';
import { CountAndProportionWithCI } from '../services/api-types';

export type EntryWithoutCI<T extends { y: CountAndProportionWithCI }> = Omit<T, 'y'> & {
  y: { count: number; proportion: number };
};

export function removeCIFromEntry<T extends { y: CountAndProportionWithCI }>(entry: T): EntryWithoutCI<T> {
  return {
    ...entry,
    y: { count: entry.y.count, proportion: entry.y.proportion.value },
  };
}
