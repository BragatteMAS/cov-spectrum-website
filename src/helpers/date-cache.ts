import assert from 'assert';
import dayjs, { Dayjs } from 'dayjs';
import { dayjsToYearWeekString, parseYearWeekString, yearWeekStringToDayjs } from './week';

export interface UnifiedDay {
  string: string;
  dayjs: Dayjs;
  isoWeek: UnifiedIsoWeek;
}

export interface UnifiedIsoWeek {
  isoYear: number;
  isoWeek: number;
  yearWeekString: string;
  firstDay: UnifiedDay;
}

class DateCache {
  private dayCache = new Map<string, UnifiedDay>();
  private isoWeekCache = new Map<string, UnifiedIsoWeek>();
  private nextWeekCache = new WeakMap<UnifiedIsoWeek, UnifiedIsoWeek>();

  getDay(dayString: string): UnifiedDay {
    return this._getDay(dayString, undefined, undefined);
  }

  getDayUsingDayjs(dayjsDay: Dayjs): UnifiedDay {
    return this.getDay(dayjsDay.format('YYYY-MM-DD'));
  }

  getIsoWeek(yearWeekString: string): UnifiedIsoWeek {
    const cachedIsoWeek = this.isoWeekCache.get(yearWeekString);
    if (cachedIsoWeek) {
      return cachedIsoWeek;
    }

    const { year, week } = parseYearWeekString(yearWeekString);
    const firstDayDayjs = yearWeekStringToDayjs(yearWeekString);
    const firstDayString = firstDayDayjs.format('YYYY-MM-DD');

    const output: UnifiedIsoWeek = {
      isoYear: year,
      isoWeek: week,
      yearWeekString,
      firstDay: (undefined as any) as UnifiedDay,
    };
    output.firstDay = this._getDay(firstDayString, firstDayDayjs, output);
    this.isoWeekCache.set(yearWeekString, output);
    return output;
  }

  rangeFromWeeks(weeks: UnifiedIsoWeek[]): { min: UnifiedIsoWeek; max: UnifiedIsoWeek } | undefined {
    let min: UnifiedIsoWeek | undefined;
    let max: UnifiedIsoWeek | undefined;
    for (const week of weeks) {
      if (!min) {
        min = week;
      } else if (this.weekIsBefore(week, min)) {
        min = week;
      }
      if (!max) {
        max = week;
      } else if (this.weekIsBefore(max, week)) {
        max = week;
      }
    }
    assert.strictEqual(min === undefined, max === undefined);
    return min && max && { min, max };
  }

  weeksFromRange(range: { min: UnifiedIsoWeek; max: UnifiedIsoWeek } | undefined): UnifiedIsoWeek[] {
    if (!range) {
      return [];
    }

    const { min, max } = range;
    assert(this.weekIsBefore(min, max) || min === max);

    const output: UnifiedIsoWeek[] = [min];
    let last = min;
    while (last !== max) {
      const next = this.getNextWeek(last);
      output.push(next);
      last = next;
    }
    return output;
  }

  // a < b
  private weekIsBefore(a: UnifiedIsoWeek, b: UnifiedIsoWeek) {
    return a.isoYear < b.isoYear || (a.isoYear === b.isoYear && a.isoWeek < b.isoWeek);
  }

  private getNextWeek(week: UnifiedIsoWeek): UnifiedIsoWeek {
    let nextWeek = this.nextWeekCache.get(week);
    if (nextWeek) {
      return nextWeek;
    }

    // this is good enough for most weeks of the year
    nextWeek = this.isoWeekCache.get(`${week.isoYear}-${week.isoWeek + 1}`);

    // cover the case where "week.isoWeek + 1" is out of range at the end of the year
    if (!nextWeek) {
      const firstDayOfNextWeek = week.firstDay.dayjs.endOf('isoWeek').add(1, 'day').startOf('isoWeek');
      nextWeek = this.getDayUsingDayjs(firstDayOfNextWeek).isoWeek;
    }

    assert(this.weekIsBefore(week, nextWeek));
    this.nextWeekCache.set(week, nextWeek);
    return nextWeek;
  }

  private _getDay(
    dayString: string,
    _dayDayjs: Dayjs | undefined,
    _isoWeek: UnifiedIsoWeek | undefined
  ): UnifiedDay {
    const cachedDay = this.dayCache.get(dayString);
    if (cachedDay) {
      return cachedDay;
    }

    assert(dayString.match(/^\d{4}-\d{2}-\d{2}$/));

    const dayDayjs = _dayDayjs || dayjs(dayString, 'YYYY-MM-DD');
    const isoWeek = _isoWeek || this.getIsoWeek(dayjsToYearWeekString(dayDayjs));
    const output: UnifiedDay = {
      string: dayString,
      dayjs: dayDayjs,
      isoWeek,
    };
    this.dayCache.set(dayString, output);
    return output;
  }
}

// It is important that there is only one DateCache instance, so that all UnifiedDay (etc.)
// objects are reference equal when they describe the same day
export const globalDateCache = new DateCache();
