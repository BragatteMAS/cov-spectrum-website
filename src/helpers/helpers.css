.dropDownButton {
  max-width: 80vw;
  margin: 10px auto 20px auto;
}
