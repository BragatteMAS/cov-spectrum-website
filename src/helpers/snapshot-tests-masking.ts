// Some external components draw from randomness and create non-deterministic output. The random parts have to be
// removed to make the snapshot tests work. This file contains common masking functions.

const uuidRegex = /[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}/gi;

export function maskUuid(obj: any) {
  if (obj) {
    for (let [key, value] of Object.entries(obj)) {
      if (typeof value === 'string' || value instanceof String) {
        obj[key] = (value as string).replaceAll(uuidRegex, '***masked***');
      } else {
        maskUuid(value);
      }
    }
  }
}
