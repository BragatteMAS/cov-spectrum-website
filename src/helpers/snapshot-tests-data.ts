import { DateCountSampleDataset } from '../data/sample/DateCountSampleDataset';
import { SamplingStrategy } from '../data/SamplingStrategy';
import { SpecialDateRangeSelector } from '../data/DateRangeSelector';
import { globalDateCache } from './date-cache';

export const dataset0: {
  variantSampleSet: DateCountSampleDataset;
  wholeSampleSet: DateCountSampleDataset;
} = {
  variantSampleSet: {
    selector: {
      location: { country: 'Germany' },
      samplingStrategy: SamplingStrategy.AllSamples,
      dateRange: new SpecialDateRangeSelector('Past6M'),
      variant: { pangoLineage: 'B.1.1.529*' },
    },
    payload: [
      { date: '2021-11-22', count: 1 },
      { date: '2021-11-23', count: 1 },
      { date: '2021-11-24', count: 3 },
      { date: '2021-11-25', count: 1 },
      { date: '2021-11-26', count: 10 },
      { date: '2021-11-27', count: 7 },
      { date: '2021-11-28', count: 10 },
      { date: '2021-11-29', count: 32 },
      { date: '2021-11-30', count: 21 },
      { date: '2021-12-01', count: 16 },
      { date: '2021-12-02', count: 18 },
      { date: '2021-12-03', count: 16 },
      { date: '2021-12-04', count: 23 },
      { date: '2021-12-05', count: 5 },
      { date: '2021-12-06', count: 21 },
      { date: '2021-12-07', count: 25 },
      { date: '2021-12-08', count: 11 },
      { date: '2021-12-09', count: 22 },
      { date: '2021-12-10', count: 12 },
      { date: '2021-12-11', count: 14 },
      { date: '2021-12-12', count: 1 },
      { date: '2021-12-13', count: 6 },
      { date: '2021-12-14', count: 7 },
      { date: '2021-12-15', count: 11 },
      { date: '2021-12-16', count: 6 },
    ].map(x => ({ ...x, date: globalDateCache.getDay(x.date) })),
  },
  wholeSampleSet: {
    selector: {
      location: { country: 'Germany' },
      samplingStrategy: SamplingStrategy.AllSamples,
      dateRange: new SpecialDateRangeSelector('Past6M'),
    },
    payload: [
      { date: '2021-07-19', count: 323 },
      { date: '2021-10-30', count: 513 },
      { date: '2021-10-24', count: 261 },
      { date: '2021-09-18', count: 787 },
      { date: '2021-09-06', count: 1596 },
      { date: '2021-09-26', count: 268 },
      { date: '2021-11-13', count: 793 },
      { date: '2021-11-23', count: 1737 },
      { date: '2021-11-24', count: 1694 },
      { date: '2021-09-03', count: 1585 },
      { date: '2021-07-05', count: 302 },
      { date: '2021-09-02', count: 1765 },
      { date: '2021-10-18', count: 1422 },
      { date: '2021-10-10', count: 182 },
      { date: '2021-07-12', count: 304 },
      { date: '2021-11-05', count: 1767 },
      { date: '2021-11-01', count: 672 },
      { date: '2021-11-16', count: 2743 },
      { date: '2021-08-22', count: 677 },
      { date: '2021-08-19', count: 984 },
      { date: '2021-08-03', count: 534 },
      { date: '2021-06-29', count: 268 },
      { date: '2021-08-23', count: 1698 },
      { date: '2021-09-25', count: 733 },
      { date: '2021-08-16', count: 1303 },
      { date: '2021-11-11', count: 1482 },
      { date: '2021-08-25', count: 1445 },
      { date: '2021-07-08', count: 208 },
      { date: '2021-12-01', count: 1335 },
      { date: '2021-07-02', count: 182 },
      { date: '2021-08-05', count: 709 },
      { date: '2021-11-21', count: 312 },
      { date: '2021-08-04', count: 618 },
      { date: '2021-10-01', count: 1002 },
      { date: '2021-07-17', count: 184 },
      { date: '2021-09-07', count: 2050 },
      { date: '2021-12-02', count: 1522 },
      { date: '2021-07-13', count: 268 },
      { date: '2021-11-15', count: 2355 },
      { date: '2021-11-22', count: 2122 },
      { date: '2021-08-02', count: 462 },
      { date: '2021-11-09', count: 1778 },
      { date: '2021-10-31', count: 449 },
      { date: '2021-09-28', count: 1769 },
      { date: '2021-10-07', count: 1065 },
      { date: '2021-10-28', count: 1272 },
      { date: '2021-10-29', count: 1183 },
      { date: '2021-09-21', count: 1510 },
      { date: '2021-08-30', count: 1336 },
      { date: '2021-11-29', count: 2576 },
      { date: '2021-12-12', count: 49 },
      { date: '2021-10-02', count: 457 },
      { date: '2021-11-18', count: 2535 },
      { date: '2021-10-25', count: 1544 },
      { date: '2021-07-07', count: 251 },
      { date: '2021-08-15', count: 356 },
      { date: '2021-09-20', count: 2199 },
      { date: '2021-10-27', count: 980 },
      { date: '2021-10-26', count: 1728 },
      { date: '2021-11-10', count: 1338 },
      { date: '2021-07-16', count: 307 },
      { date: '2021-11-08', count: 2571 },
      { date: '2021-08-29', count: 413 },
      { date: '2021-11-25', count: 1554 },
      { date: '2021-10-17', count: 454 },
      { date: '2021-11-07', count: 168 },
      { date: '2021-08-27', count: 1633 },
      { date: '2021-12-06', count: 1744 },
      { date: '2021-07-06', count: 202 },
      { date: '2021-12-16', count: 124 },
      { date: '2021-08-09', count: 951 },
      { date: '2021-09-13', count: 1661 },
      { date: '2021-11-26', count: 2088 },
      { date: '2021-12-07', count: 1070 },
      { date: '2021-12-11', count: 328 },
      { date: '2021-08-11', count: 967 },
      { date: '2021-11-28', count: 302 },
      { date: '2021-11-14', count: 305 },
      { date: '2021-10-20', count: 1118 },
      { date: '2021-11-02', count: 1107 },
      { date: '2021-10-12', count: 1192 },
      { date: '2021-09-14', count: 1982 },
      { date: '2021-08-24', count: 2056 },
      { date: '2021-06-28', count: 300 },
      { date: '2021-08-17', count: 1317 },
      { date: '2021-09-09', count: 1657 },
      { date: '2021-10-15', count: 753 },
      { date: '2021-06-27', count: 44 },
      { date: '2021-08-14', count: 564 },
      { date: '2021-09-11', count: 991 },
      { date: '2021-11-03', count: 1733 },
      { date: '2021-09-29', count: 937 },
      { date: '2021-10-23', count: 1131 },
      { date: '2021-07-28', count: 410 },
      { date: '2021-10-21', count: 951 },
      { date: '2021-09-24', count: 1655 },
      { date: '2021-10-13', count: 737 },
      { date: '2021-11-17', count: 1310 },
      { date: '2021-08-20', count: 1529 },
      { date: '2021-08-28', count: 732 },
      { date: '2021-07-25', count: 79 },
      { date: '2021-07-31', count: 229 },
      { date: '2021-07-15', count: 320 },
      { date: '2021-09-19', count: 351 },
      { date: '2021-10-14', count: 836 },
      { date: '2021-11-20', count: 960 },
      { date: '2021-11-04', count: 1699 },
      { date: '2021-08-21', count: 758 },
      { date: '2021-09-10', count: 1632 },
      { date: '2021-12-05', count: 273 },
      { date: '2021-07-23', count: 412 },
      { date: '2021-10-04', count: 1451 },
      { date: '2021-11-12', count: 1132 },
      { date: '2021-07-10', count: 134 },
      { date: '2021-10-06', count: 793 },
      { date: '2021-09-23', count: 1074 },
      { date: '2021-12-09', count: 1348 },
      { date: '2021-11-19', count: 1169 },
      { date: '2021-08-31', count: 1916 },
      { date: '2021-12-03', count: 1250 },
      { date: '2021-10-22', count: 1163 },
      { date: '2021-07-04', count: 42 },
      { date: '2021-12-10', count: 765 },
      { date: '2021-12-04', count: 1058 },
      { date: '2021-11-27', count: 696 },
      { date: '2021-09-22', count: 1146 },
      { date: '2021-08-10', count: 944 },
      { date: '2021-10-09', count: 532 },
      { date: '2021-06-22', count: 254 },
      { date: '2021-08-13', count: 835 },
      { date: '2021-10-05', count: 1552 },
      { date: '2021-10-19', count: 1889 },
      { date: '2021-09-17', count: 1489 },
      { date: '2021-06-26', count: 93 },
      { date: '2021-07-18', count: 138 },
      { date: '2021-07-27', count: 520 },
      { date: '2021-09-08', count: 2428 },
      { date: '2021-09-15', count: 1314 },
      { date: '2021-06-25', count: 178 },
      { date: '2021-08-12', count: 838 },
      { date: '2021-07-21', count: 552 },
      { date: '2021-11-30', count: 2153 },
      { date: '2021-11-06', count: 980 },
      { date: '2021-12-13', count: 40 },
      { date: '2021-08-08', count: 129 },
      { date: '2021-12-08', count: 424 },
      { date: '2021-08-01', count: 192 },
      { date: '2021-09-04', count: 1146 },
      { date: '2021-07-22', count: 557 },
      { date: '2021-09-16', count: 1606 },
      { date: '2021-10-08', count: 1046 },
      { date: '2021-06-21', count: 195 },
      { date: '2021-10-16', count: 557 },
      { date: '2021-12-14', count: 72 },
      { date: '2021-07-11', count: 50 },
      { date: '2021-07-09', count: 226 },
      { date: '2021-09-05', count: 398 },
      { date: '2021-09-01', count: 1701 },
      { date: '2021-06-24', count: 174 },
      { date: '2021-07-29', count: 441 },
      { date: '2021-09-27', count: 1502 },
      { date: '2021-10-03', count: 326 },
      { date: '2021-07-30', count: 543 },
      { date: '2021-12-17', count: 5 },
      { date: '2021-06-30', count: 166 },
      { date: '2021-07-14', count: 262 },
      { date: '2021-06-23', count: 175 },
      { date: '2021-07-01', count: 207 },
      { date: '2021-08-18', count: 1198 },
      { date: '2021-10-11', count: 1236 },
      { date: '2021-08-07', count: 431 },
      { date: '2021-08-06', count: 534 },
      { date: '2021-07-26', count: 445 },
      { date: '2021-09-30', count: 1180 },
      { date: '2021-08-26', count: 1460 },
      { date: '2021-09-12', count: 379 },
      { date: '2021-07-24', count: 426 },
      { date: '2021-07-20', count: 456 },
      { date: '2021-07-03', count: 113 },
      { date: '2021-12-15', count: 94 },
    ].map(x => ({ ...x, date: globalDateCache.getDay(x.date) })),
  },
};
