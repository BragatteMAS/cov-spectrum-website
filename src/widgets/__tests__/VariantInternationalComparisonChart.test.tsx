import 'jest-canvas-mock';
import ResizeObserver from 'resize-observer-polyfill';
import renderer, { act } from 'react-test-renderer';
import { maskRegex, maskUuid } from '../../helpers/snapshot-tests-masking';
import React from 'react';
import { dataset0 } from '../../helpers/snapshot-tests-data';
import { VariantInternationalComparisonChart } from '../VariantInternationalComparisonChart';

window.ResizeObserver = ResizeObserver;

describe('<VariantInternationalComparisonChart>', () => {
  it('dataset0 renders correctly', async () => {
    const { variantInternationalDateCount, wholeInternationalDateCount } = dataset0;
    const tree = renderer.create(
      <VariantInternationalComparisonChart
        preSelectedCountries={['Germany']}
        variantInternationalSampleSet={variantInternationalDateCount}
        wholeInternationalSampleSet={wholeInternationalDateCount}
      />
    );
    await act(async () => {});
    const snapshot = tree.toJSON();
    maskUuid(snapshot);
    maskRegex(snapshot, /css-[0-9a-z]+/gi); // Mask weird css class names generated by react-select
    expect(snapshot).toMatchSnapshot();
  });
});
