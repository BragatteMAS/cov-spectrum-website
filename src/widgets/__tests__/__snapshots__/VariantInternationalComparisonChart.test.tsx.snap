// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VariantInternationalComparisonChart> dataset0 renders correctly 1`] = `
<div
  className="relative h-full"
>
  <div
    className="sc-bcXHqe idcAl"
  >
    <div
      className="sc-hLBbgP kiIjgh"
    >
      <div
        className="mt-2 mb-4 w-40"
      >
        <button
          className="undefined bg-gray-400 hover:bg-gray-500 focus:ring-gray-500 focus:ring-offset-gray-200 px-2 text-white w-full transition ease-in duration-100 scale-110 text-center text-base font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2 rounded-lg "
          onClick={[Function]}
        >
          Toggle log scale
        </button>
      </div>
      <div
        className=" ***masked***-container"
        onKeyDown={[Function]}
      >
        <span
          className="***masked***-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="***masked***-a11yText-A11yText"
        />
        <div
          className=" ***masked***-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" ***masked***-Component"
          >
            <div
              className=" ***masked***-multiValue"
            >
              <div
                className=" ***masked***-Component"
              >
                Germany
              </div>
              <div
                aria-label="Remove Germany"
                className=" ***masked***-Component"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                role="button"
              >
                <svg
                  aria-hidden="true"
                  className="***masked***-Svg"
                  focusable="false"
                  height={14}
                  viewBox="0 0 20 20"
                  width={14}
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
            </div>
            <div
              className=" ***masked***-multiValue"
            >
              <div
                className=" ***masked***-Component"
              >
                United Kingdom
              </div>
              <div
                aria-label="Remove United Kingdom"
                className=" ***masked***-Component"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                role="button"
              >
                <svg
                  aria-hidden="true"
                  className="***masked***-Svg"
                  focusable="false"
                  height={14}
                  viewBox="0 0 20 20"
                  width={14}
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
            </div>
            <div
              className=" ***masked***-multiValue"
            >
              <div
                className=" ***masked***-Component"
              >
                United States
              </div>
              <div
                aria-label="Remove United States"
                className=" ***masked***-Component"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                role="button"
              >
                <svg
                  aria-hidden="true"
                  className="***masked***-Svg"
                  focusable="false"
                  height={14}
                  viewBox="0 0 20 20"
                  width={14}
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
            </div>
            <div
              className=" ***masked***-multiValue"
            >
              <div
                className=" ***masked***-Component"
              >
                Denmark
              </div>
              <div
                aria-label="Remove Denmark"
                className=" ***masked***-Component"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                role="button"
              >
                <svg
                  aria-hidden="true"
                  className="***masked***-Svg"
                  focusable="false"
                  height={14}
                  viewBox="0 0 20 20"
                  width={14}
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
            </div>
            <div
              className=" ***masked***-multiValue"
            >
              <div
                className=" ***masked***-Component"
              >
                South Africa
              </div>
              <div
                aria-label="Remove South Africa"
                className=" ***masked***-Component"
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                role="button"
              >
                <svg
                  aria-hidden="true"
                  className="***masked***-Svg"
                  focusable="false"
                  height={14}
                  viewBox="0 0 20 20"
                  width={14}
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
            </div>
            <div
              className=" ***masked***-Component"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" ***masked***-Component"
          >
            <div
              aria-hidden="true"
              className=" ***masked***-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="***masked***-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              className=" ***masked***-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" ***masked***-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="***masked***-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="recharts-responsive-container"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
